### PR TITLE
Add DPI parameter on SaveToPNG function

### PR DESCRIPTION
**Pull request by diegobernardes**
The default, and minimum, DPI has been raised to 96. The client can choose a bigger value.

Issue: CW-608

### CODE REVIEW

File: main.c [hunk 1]
```diff
@@ -249,6 +249,10 @@ save_to_png_output save_to_png(save_to_png_input input) {
 			}
 		}
 
+		// Update the scale_factor based on the DPI.
+		float scale_dpi = ((float)(input.dpi - 72) / 72) * 100;
+		scale_factor = scale_factor * (1 + (scale_dpi / 100));
+
```
Comment (rlasskiy) [inline main.c:255]: We should avoid doing manual calculations. It's easy to make mistakes and hard to understand during code reviews.

Looking to library implementation to understand how to do this right

Comment (diegobernardes) [reply main.c:255]: > (input.dpi - 72) / 72 looks wrong for me

This produces a percentage increase based on the DPI. `(96-72)/72 = 0,33` This means that we're applying a 33% increase at the scale factor calculated on previous steps.

Comment (rlasskiy) [reply main.c:255]: So it should be a simple matrix operation like:
```
fz_matrix  dpi_matrix = fz_scale(input.dpi / 72, input.dpi / 72);
fz_matrix  scale_matrix = fz_scale(scale_factor, scale_factor);
fz_matrix  page_matrix = fz_concat(dpi_matrix, scale_matrix);
```

which could be simplified to something like
```
fz_matrix  page_matrix = fz_concat(
    fz_scale(input.dpi / 72, input.dpi / 72),
    fz_scale(scale_factor, scale_factor)
);
```

Comment (diegobernardes) [reply main.c]: Done, could you check again?

**Review by rlasskiy — APPROVED** [review]

**Pull request merged**